python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1303 @@
+                                                        afwMath.MEANCLIP, sctrl).getValue()
+                returnAreas[ampName] = rescaleIm
+
+        if level == 'DETECTOR':  # else just average the whole detector
+            # xxx I think temp can be done away with by just adding the mean here, right?
+            detName = local_exp.getDetector().getId()
+            means[detName] = afwMath.makeStatistics(temp[border:-border, border:-border],
+                                                    afwMath.MEANCLIP, sctrl).getValue()
+            returnAreas[detName] = rescaleIm
+
+        return returnAreas, means
+
+    def _crossCorrelate(self, maskedIm0, maskedIm1, frameId=None, detId=None):
+        """Calculate the cross-correlation of an area.
+
+        If the areas in question contains many amplifiers then these must have been gain corrected.
+
+        Parameters:
+        -----------
+        area1 : `lsst.afw.image.MaskedImageF`
+            The first image area
+        area2 : `lsst.afw.image.MaskedImageF`
+            The first image area
+        frameId : `str`, optional
+            The frame identifier for use in the filename if writing debug outputs.
+        detId : `str`, optional
+            The detector identifier (detector, or detector+amp depending on level)
+            for use in the filename if writing debug outputs.
+
+        Returns:
+        --------
+        xcorr : `np.ndarray`
+            The quarter-image cross-correlation
+        mean : `float`
+            The as-calculated means of the input images (clipped, and with borders applied)
+
+        Notes:
+        ------
+        This function is controlled by the following pexConfig parameters:
+        maxLag : `int`
+            The maximum lag to use in the cross-correlation calculation
+        nPixBorderXCorr : `int`
+            The number of border pixels to exclude
+        nSigmaClipXCorr : `float`
+            The number of sigma to be clipped to
+        biasCorr : `float`
+            Parameter used to correct from the bias introduced by the sigma cuts.
+        """
+        maxLag = self.config.maxLag
+        border = self.config.nPixBorderXCorr
+        sigma = self.config.nSigmaClipXCorr
+        biasCorr = self.config.biasCorr
+
+        sctrl = afwMath.StatisticsControl()
+        sctrl.setNumSigmaClip(sigma)
+
+        # Diff the images, and apply border
+        diff = maskedIm0.clone()
+        diff -= maskedIm1.getImage()
+        diff = diff[border:-border, border:-border]
+
+        if self.debug.writeDiffImages:
+            filename = '_'.join(['diff', 'detector', detId, frameId, '.fits'])
+            diff.writeFits(os.path.join(self.debug.debugDataPath, filename))
+
+        # Subtract background.  It should be a constant, but it isn't always
+        # xxx do we want some logic here for whether to subtract or not? Or a config option?
```

Comment:
Any chance of figuring this out before merging? If not, please file a ticket and reference it here.

Reply:
I have made it log warnings the background is found to be above a configurable level, so people in the future will know if it's subtracting something large. I think this should do it.